### PR TITLE
update gogo/protobuf dependency

### PR DESCRIPTION
**Pull request by jmuk**
- update to the head of v0.5, resulting with code changes
- the version now supports unmarshaling of Struct
- interestingly Gopkg.lock does not require this change, it
  already points to the same revision

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -704,11 +704,13 @@ go_repository(
     importpath = "github.com/go-openapi/swag",
 )
 
-go_repository(
-    name = "com_github_gogo_protobuf",
-    commit = "c0656edd0d9eab7c66d1eb0c568f9039345796f7",
-    importpath = "github.com/gogo/protobuf",
-)
+# gogo/protobuf dependency is described in mixer/adapter_author_deps.bzl.
```
Comment (douglas-reid) [inline WORKSPACE]: i think it would be OK to just remove this entirely at this point.

Comment (jmuk) [reply WORKSPACE]: I think it would be helpful for future updaters to find out the right place...

**Pull request merged**